assets/src/scripts/post-list-filters.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Toggles the post list filter container.
+ *
+ * @param {HTMLElement} filterButton - The button used to toggle the filter container.
+ * @param {HTMLElement} filterContainer - The filter container.
+ */
+const toggleFilterContainer = ( filterButton, filterContainer ) => {
+	/**
+	 * Click event handler for the filter button.
+	 */
+	const clickHandler = () => {
+		const isHidden = ! filterContainer.classList.contains( 'post-list-filter__container--open' );
+		filterContainer.classList.toggle( 'post-list-filter__container--open' );
+
+		filterButton.innerHTML = isHidden ? 'Hide filters' : filterButton.dataset.showFiltersAppliedButtonLabel;
```

Comment:
This string still needs some kind of internationalization, so it can be displayed in non-English languages.